project1/client/client_tasks/download_task.h

```diff
@@ -0,0 +1,34 @@
+#ifndef PROJECT1_DOWNLOAD_TASK_H
+#define PROJECT1_DOWNLOAD_TASK_H
+
+#include "../../thread_pool/task.h"
+#include "../../wire_protocol/wire_protocol.h"
+#include "../client_util.h"
+
+namespace client_tasks {
+class DownloadTask : public thread_pool::Task {
+ public:
+  DownloadTask(const std::string& filename, std::vector<uint8_t> &incoming_buf,
+               size_t buf_size, int client_fd,
+               wire_protocol::MessageParser<google::protobuf::Message> &parser
+               ) {
```

Comment:
(minor) Prefer initialization list syntax to initialize class members. Also, any reason this implementation isn't in the .cpp file?

Reply:
Can you expand more on what you mean by list syntax? I'm not familiar.

I did not know where best instantiate this. Should it be in the .cpp?

Reply:
I'm referring to [this](https://en.cppreference.com/w/cpp/language/constructor). In the way of examples, pretty much all the other classes in this repo that take constructor arguments use this strategy. I usually do put it in the .cpp file.